compiler/luci-interpreter/src/kernels/FullyConnected.cpp

```diff
@@ -235,14 +235,47 @@ void FullyConnected::evalHybridWI4AF32() const
   params.float_activation_max = activation_max;
   params.weights_format = tflite::FullyConnectedWeightsFormat::kDefault;
 
-  const int8_t *weights_int4 = getTensorData<int8_t>(weights());
-  float *weights_float = getTensorData<float>(scratch());
+  const auto *weights_int4 = getTensorData<int8_t>(weights());
+  auto *weights_float = getTensorData<float>(scratch());
   const Shape &weights_shape = weights()->shape();
-  for (int32_t i = 0; i < weights_shape.num_elements(); ++i)
+  const auto weights_scales = weights()->scales();
+  const auto weights_quantized_dimension = weights()->quantized_dimension();
+  LUCI_INTERPRETER_CHECK(weights_quantized_dimension == 0);
```

Comment:
I think there there should be some comment for `weights_quantized_dimension == 0`
Is this for permanent thing? or can be changed in the future?

Reply:
This is an invariant for per-channel quantization of FC weights.